File: 1978.c
#include<stdio.h>

int main(){
    int numOfCases = 0;
    int cases[100] = {0, };
    int counter = 0;
    scanf("%d", &numOfCases);
    for(int i = 0; i < numOfCases; i++){
        scanf("%d", &cases[i]);
    }
    for(int i = 0; i < numOfCases; i++){
        int j = 2;
        while(1){
            if(cases[i] == 1){
                //printf("@1 detected! (not prime)\n");
                break;
            }
            else if(j == cases[i]){
                //printf("@%d is prime number. it only includes itself!\n", j);
                counter++;
                break;
            }
            else if(cases[i] % j == 0){
                //printf("@%d is not prime numbeer. it is %d x %d = %d\n", cases[i], j, cases[i]/j, cases[i]);
                break;
            }
            else if(j > cases[i])
                break;
            j++;
        }
    }
    printf("%d", counter);
    return 0;
}

File: 1978.java
import java.io.*;
import java.util.StringTokenizer;

public class Main{
    public static void main(String[] args) throws IOException{
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
        int numOfCases = Integer.parseInt(br.readLine());
        int counter = 0;
        String tmp = br.readLine();
        StringTokenizer st = new StringTokenizer(tmp);
        for(int i = 0; i < numOfCases; i++){
            int inputCase = Integer.parseInt(st.nextToken());
            int j = 2;
            while(true){
                if(inputCase == 1){
                    //System.out.println("@ 1 is NOT prime number");
                    break;
                }
                else if(inputCase == j){
                    //System.out.println("@ "+j+" only includes itself. it is prime number");
                    counter++;
                    break;
                }
                else if(inputCase % j == 0){
                    //System.out.println("@ "+inputCase+" is NOT prime number. it could represented like : "+j+" X "+(inputCase/j));
                    break;
                }
                else if(j > inputCase){
                    //System.out.println("@ "+inputCase+" is NOT prime number.");
                    break;
                }
                j++;
            }
        }
        System.out.println(counter);
    }
}
